sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/PartitionKeyBuilder.java

```diff
@@ -52,15 +52,56 @@ public PartitionKeyBuilder add(boolean value) {
         return this;
     }
 
+    /**
+     * Adds a null partition key value
+     * @return The current PartitionKeyBuilder object
+     * @deprecated Null value should only be used with PartitionKey constructor.
+     */
+    @Deprecated
+    public PartitionKeyBuilder addNullValue() {
+        this.partitionKeyValues.add(null);
+        return this;
+    }
+
+    /**
+     * Adds a None Partition Key value to the path. An error will be raised if used with other paths.
+     * @return The current PartitionKeyBuilder object
+     * @deprecated PartitionKey.None value should only be used with PartitionKey constructor.
+     */
+    @Deprecated
+    public PartitionKeyBuilder addNoneValue() {
+        this.partitionKeyValues.add(PartitionKey.NONE);
+        return this;
+    }
+
     /**
      * Builds a new instance of the type PartitionKey with the specified Partition Key values.
      * @return PartitionKey object
+     * @throws IllegalStateException when using PartitionKey.None with other values
      */
     public PartitionKey build() {
+        // Why these checks?
+        // These changes are being added for SDK to support multiple paths in a partition key.
+        //
+        // Currently, when a resource does not specify a value for the PartitionKey,
+        // we assign a temporary value `PartitionKey.None` and later discern whether
+        // it is a PartitionKey.Undefined or PartitionKey.Empty based on the Collection Type.
+        // We retain this behaviour for single path partition keys.
+        //
+        // For collections with multiple path keys, absence of a partition key values is
+        // always treated as a PartitionKey.Undefined.
         if(this.partitionKeyValues.size() == 0) {
             throw new IllegalArgumentException("No partition key value has been specified");
         }
 
+        if(this.partitionKeyValues.size() == 1 && PartitionKey.NONE.equals(this.partitionKeyValues.get(0))) {
```

Comment:
will this logic be safe to remove when we delete the deprecated API?

Reply:
it will since none of the other add methods allow the PartitionKeyBuilder to take in a PartitionKey.None